tests/models/musicgen/test_modeling_musicgen.py

```diff
@@ -189,6 +198,47 @@ def setUp(self):
     def test_config(self):
         self.config_tester.run_common_tests()
 
+    # special case for labels
+    def _prepare_for_class(self, inputs_dict, model_class, return_labels=False):
+        inputs_dict = super()._prepare_for_class(inputs_dict, model_class, return_labels=return_labels)
+
+        if return_labels:
+            inputs_dict["labels"] = torch.zeros(
+                (self.model_tester.batch_size, self.model_tester.seq_length, self.model_tester.num_codebooks),
+                dtype=torch.long,
+                device=torch_device,
+            )
+        else:
+            inputs_dict.pop("labels", None)
+        return inputs_dict
+
+    def check_training_gradient_checkpointing(self, gradient_checkpointing_kwargs=None):
+        if not self.model_tester.is_training:
+            return
+
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+        config.use_cache = False
+        config.return_dict = True
+        model = MusicgenForCausalLM(config)
+
+        model.to(torch_device)
+        model.gradient_checkpointing_enable(gradient_checkpointing_kwargs=gradient_checkpointing_kwargs)
+        model.train()
+
+        # Contrarily to the initial method, we don't unfreeze freezed parameters.
+        # Otherwise, it'll mess with the freezed sinusoidal embeddings
```

Comment:
It'll change the weights of the pre-trained embeddings for sure, but the code should still run right? Or is there an issue where the code won't run if we train the embeddings? Unless this is the case, I would just use the base `check_training_gradient_checkpointing` method for simplicity

Reply:
The comments are not really clear, I'll make it clearer: the ConditionalGeneration models have the audio encoder which is never used outside of the `.generate`, if we don't freeze it, we'll have some issues because trainable weights won't have seen any gradients.